### ISSUE TITLE
static key mode

### ISSUE THREAD

**Issue by hannesm** [roburio/openvpn#11]
openvpn has a "simple, static key mode" -- where no ephemeral handshake is done, but pre-shared keys are used (requires an established secure channel between both parties). this means: no TLS handshake, ... . how this is implemented is:
- data is just sent to the remote, encrypted and authenticated (i.e. 20 bytes hash, 16 bytes iv, rest data)
  - data contains packed id (4 byte), time_t (4 byte), 0xfa (if no lzo), followed by the actual data
- for configuration, `OCC` (the OpenVPN configuration protocol) is used (data with 16 byte hardcoded cookie, followed by single byte command, and (optional) payload). here, a client may request the server config, discover MTU, ... (see occ.h in openvpn) -- which appears to be optional (since they broke it at 1.5 beta 8)

TL;DR: an implementation of static keys is pretty straightforward, esp. neglecting OCC. Of course, supporting OCC may be worth doing (some - such as mtu and load test and exit - may as well be used for OpenVPN-with-TLS-handshake). The question (for @cfcs) is mainly whether we want support for this, and if so, whether printng a big warning is sufficient (or should we introduce a "--i-realy-dont-need-forward-secrecy" command line argument)?

**Comment by cfcs** [comment]
Hmm, unless we have a clear use-case for static keys I don't see much point in going out of our way to support it, since as you mention it is kind of bad.

If we do have concrete use-cases, I think a warning would be fine (no need to mandate a `--im-using-insecure-stuff`)

**Comment by hannesm** [comment]
well, the only reason i care about it is that i have a use-case -- i got an openvpn configuration from a friend (i.e. i don't have control over the other end), and that one uses static key and udp mode.

**Comment by cfcs** [comment]
Cool, then I think it makes sense to support it :)
